### PR TITLE
Enable sanitizers on CIFuzz.

### PR DESCRIPTION
**Pull request by aouinizied**
Current CI Fuzz runs with default sanitizer which is "address".
In this PR, we enable all available sanitizers as a matrix.

Zied

### CODE REVIEW

**Comment by utoni** [comment]
Shouldn't the asan fuzzing job fail?

**Comment by aouinizied** [comment]
@IvanNardi @lnslbrty Yes, this is weird.
I checked  the logs and we have errors and artifacts are created, so do not know why Run Fuzzers always pass.
I added a temporary hack (removed failure condition) to be able to have artifacts on all cases, so after this PR at least you will have the pcaps :).

**Comment by IvanNardi** [comment]
@lnslbrty , here you are the QUIC pcap triggering the crash!
[quic-fuzz-crash.zip](https://github.com/ntop/nDPI/files/7322584/quic-fuzz-crash.zip)

Crash confirmed on current master

**Comment by aouinizied** [comment]
pr_fuzz

And now we have all artifacts uploaded each time the CI is triggered. One artifact by sanitizer.

Zied

**Comment by IvanNardi** [comment]
> @IvanNardi @lnslbrty Yes, this is weird. I checked the logs and we have errors and artifacts are created, so do not know why Run Fuzzers always pass. I added a temporary hack (removed failure condition) to be able to have artifacts on all cases, so after this PR at least you will have the pcaps :).

I confirm that the jobs don't fail but we have the artifacts!
Thank you very much. very useful!!-

**Comment by utoni** [comment]
Thank you both! Mercí beaucoup and Mille grazie. :smile:

File: .github/workflows/cifuzz.yml [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
-       dry-run: false
-   - name: Run Fuzzers
-     uses: google/oss-fuzz/infra/cifuzz/actions/run_fuzzers@master
-     with:
-       oss-fuzz-project-name: 'ndpi'
-       fuzz-seconds: 600
-       dry-run: false
-   - name: Upload Crash
-     uses: actions/upload-artifact@v1
-     if: failure()
```
Comment (IvanNardi) [inline .github/workflows/cifuzz.yml:20]: @aouinizied, could you add a comment here about why we disabled this statement, please? So we don't forget it, hopefully....

**Comment by aouinizied** [comment]
@IvanNardi Added the comment.
I will try to add a step where we count crash artifacts and if they are greater than 0, trigger failure and make uploading conditioned by this failure. It's a more clean workaround.

**Comment by aouinizied** [comment]
@IvanNardi @lnslbrty I fixed this strange behavior by adding a step that checks manually for crashes. 
As GitHub action use the exit code as status. Now the condition if failure() will work as expected when uploading crashes.
Ready to merge even if Fuzz CIs will fail as expected :).

Zied

**Comment by IvanNardi** [comment]
Thanks a lot!

**Pull request merged**